Tests/SentryTests/Integrations/Performance/IO/SentryNSDataTrackerTests.swift

```diff
@@ -11,8 +11,9 @@ class SentryNSDataTrackerTests: XCTestCase {
                 
         func getSut() -> SentryNSDataTracker {
             let result = SentryNSDataTracker.sharedInstance
+            Dynamic(result).processInfoWrapper = TestProcessInfoWrapper()
```

Comment:
`m`: Please pass the process info wrapper in as a dependency to `SentryNSDataTracker:init` instead, so the initializer clearly communicates its dependencies.

Reply:
SentryNSDataTracker is used by `SentryNSDataSwizzling`, and since swizzling is permanent, the reference to `SentryNSDataTracker` will be kept by the swizzled method. 
Thats why `SentryNSDataTracker` and so many other tracker classes used with swizzling are Singletons. 
It is possible to change this approach, but imo not worth it.

Reply:
Even tho we would just changing the singleton behaviour around, maybe we should only use singleton with the swizzling classes, this classes are already only tested during integration tests.

Reply:
> and since swizzling is permanent, the reference to SentryNSDataTracker will be kept by the swizzled method.

That's already the case.

> Thats why SentryNSDataTracker and so many other tracker classes used with swizzling are Singletons.

They shouldn't know that they are singletons. Instead, the SentryDependencyContainer or another class should decide that for them. Dependency injection is key for testability and readability cause it helps you to understand which dependencies a class has. Now I need to read the code of SentryNSDataTracker and the code of its test to understand how to test it and which internal dependencies it has. IMO it's worth the refactoring.

Reply:
>>and since swizzling is permanent, the reference to SentryNSDataTracker will be kept by the swizzled method.
>
> That's already the case.

Yes, and this is why `SentryNSDataTracker` needed to be singleton. To problem was in `SentryNSDataSwizzling`

> They shouldn't know that they are singletons...

You didn't get my point. 
SentryNSDataTracker used to follow the pattern of other classes that we already have. Classes that are used by Swizzling static classes.

But this doesn't matter anymore, I change this approach for SentryNSDataTracker